File: .sling/libs/sushi/eqela/jk-core/0.33.0/jk.time.format.lua
jk = jk or {}

jk.time = jk.time or {}

jk.time.format = jk.time.format or {}

jk.time.format.VerboseDateTimeString = {}
jk.time.format.VerboseDateTimeString.__index = jk.time.format.VerboseDateTimeString
_vm:set_metatable(jk.time.format.VerboseDateTimeString, {})

function jk.time.format.VerboseDateTimeString._create()
	local v = _vm:set_metatable({}, jk.time.format.VerboseDateTimeString)
	return v
end

function jk.time.format.VerboseDateTimeString:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.time.format.VerboseDateTimeString'
	self['_isType.jk.time.format.VerboseDateTimeString'] = true
end

function jk.time.format.VerboseDateTimeString:_construct0()
	jk.time.format.VerboseDateTimeString._init(self)
	return self
end

function jk.time.format.VerboseDateTimeString:forUTCNow()
	do return _g.jk.time.format.VerboseDateTimeString:forDateTime(_g.jk.time.SystemClock:asDateTimeUTC()) end
end

function jk.time.format.VerboseDateTimeString:forNow()
	do return _g.jk.time.format.VerboseDateTimeString:forDateTime(_g.jk.time.SystemClock:asDateTimeLocal()) end
end

function jk.time.format.VerboseDateTimeString:forDateTime(dd)
	if not (dd ~= nil) then
		do return "NODATE" end
	end
	do
		local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
		do sb:appendString(_g.jk.time.format.VerboseDateTimeString:getShortDayName(dd:getWeekDay())) end
		do sb:appendString(", ") end
		do sb:appendString(_g.jk.lang.String:forInteger(dd:getDayOfMonth())) end
		do sb:appendCharacter(32) end
		do sb:appendString(_g.jk.time.format.VerboseDateTimeString:getShortMonthName(dd:getMonth())) end
		do sb:appendCharacter(32) end
		do sb:appendString(_g.jk.lang.String:forInteger(dd:getYear())) end
		do sb:appendCharacter(32) end
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getHours(), 2, "0")) end
		do sb:appendCharacter(58) end
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getMinutes(), 2, "0")) end
		do sb:appendCharacter(58) end
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getSeconds(), 2, "0")) end
		do
			local offset = dd:getTimezoneOffset()
			if offset == 0 then
				do sb:appendString(" +0000") end
			else
				local hr = _util:convert_to_integer(offset / 3600)
				local mm = _util:convert_to_integer((offset - hr * 3600) / 60)
				if hr <= 0 then
					do sb:appendString(" +") end
					hr = -hr
				else
					do sb:appendString(" -") end
				end
				do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(hr, 2, "0")) end
				do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(mm, 2, "0")) end
			end
			do return sb:toString() end
		end
	end
end

function jk.time.format.VerboseDateTimeString:getTimeStringForDateTime(dd, includeTimeZone)
	if not (dd ~= nil) then
		do return "NOTIME" end
	end
	do
		local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getHours(), 2, "0")) end
		do sb:appendCharacter(58) end
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getMinutes(), 2, "0")) end
		do sb:appendCharacter(58) end
		do sb:appendString(_g.jk.lang.String:forIntegerWithPadding(dd:getSeconds(), 2, "0")) end
		if includeTimeZone then
			do sb:appendString(" GMT") end
		end
		do return sb:toString() end
	end
end

function jk.time.format.VerboseDateTimeString:getDateStringForDateTime(dd)
	if not (dd ~= nil) then
		do return "NODATE" end
	end
	do
		local sb = _g.jk.lang.StringBuilder._construct0(_g.jk.lang.StringBuilder._create())
		do sb:appendString(_g.jk.time.format.VerboseDateTimeString:getLongMonthName(dd:getMonth())) end
		do sb:appendCharacter(32) end
		do sb:appendString(_g.jk.lang.String:forInteger(dd:getDayOfMonth())) end
		do sb:appendString(", ") end
		do sb:appendString(_g.jk.lang.String:forInteger(dd:getYear())) end
		do return sb:toString() end
	end
end

function jk.time.format.VerboseDateTimeString:getShortDayName(n)
	do
		local v = n
		if v == 1 then
			do return "Sun" end
		elseif v == 2 then
			do return "Mon" end
		elseif v == 3 then
			do return "Tue" end
		elseif v == 4 then
			do return "Wed" end
		elseif v == 5 then
			do return "Thu" end
		elseif v == 6 then
			do return "Fri" end
		elseif v == 7 then
			do return "Sat" end
		end
	end
	do return nil end
end

function jk.time.format.VerboseDateTimeString:getLongDayName(n)
	do
		local v = n
		if v == 1 then
			do return "Sunday" end
		elseif v == 2 then
			do return "Monday" end
		elseif v == 3 then
			do return "Tuesday" end
		elseif v == 4 then
			do return "Wednesday" end
		elseif v == 5 then
			do return "Thursday" end
		elseif v == 6 then
			do return "Friday" end
		elseif v == 7 then
			do return "Saturday" end
		end
	end
	do return nil end
end

function jk.time.format.VerboseDateTimeString:getShortMonthName(n)
	do
		local v = n
		if v == 1 then
			do return "Jan" end
		elseif v == 2 then
			do return "Feb" end
		elseif v == 3 then
			do return "Mar" end
		elseif v == 4 then
			do return "Apr" end
		elseif v == 5 then
			do return "May" end
		elseif v == 6 then
			do return "Jun" end
		elseif v == 7 then
			do return "Jul" end
		elseif v == 8 then
			do return "Aug" end
		elseif v == 9 then
			do return "Sep" end
		elseif v == 10 then
			do return "Oct" end
		elseif v == 11 then
			do return "Nov" end
		elseif v == 12 then
			do return "Dec" end
		end
	end
	do return nil end
end

function jk.time.format.VerboseDateTimeString:getLongMonthName(n)
	do
		local v = n
		if v == 1 then
			do return "January" end
		elseif v == 2 then
			do return "February" end
		elseif v == 3 then
			do return "March" end
		elseif v == 4 then
			do return "April" end
		elseif v == 5 then
			do return "May" end
		elseif v == 6 then
			do return "June" end
		elseif v == 7 then
			do return "July" end
		elseif v == 8 then
			do return "August" end
		elseif v == 9 then
			do return "September" end
		elseif v == 10 then
			do return "October" end
		elseif v == 11 then
			do return "November" end
		elseif v == 12 then
			do return "December" end
		end
	end
	do return nil end
end

File: .sling/libs/web5/eqela/jk-core/0.36.0/jk.time.format.js
let JkTimeFormatVerboseDateTimeString = function() {};

JkTimeFormatVerboseDateTimeString.prototype._t = { "JkTimeFormatVerboseDateTimeString" : true };
JkTimeFormatVerboseDateTimeString.prototype._tobj = JkTimeFormatVerboseDateTimeString;

JkTimeFormatVerboseDateTimeString.NEW = function() {
	var v = new JkTimeFormatVerboseDateTimeString;
	return v.CTOR_JkTimeFormatVerboseDateTimeString();
};

JkTimeFormatVerboseDateTimeString.prototype.CTOR_JkTimeFormatVerboseDateTimeString = function() {
	return this;
};

JkTimeFormatVerboseDateTimeString.forUTCNow = function() {
	return JkTimeFormatVerboseDateTimeString.forDateTime((JkTimeSystemClock.asDateTimeUTC()));
};

JkTimeFormatVerboseDateTimeString.forNow = function() {
	return JkTimeFormatVerboseDateTimeString.forDateTime((JkTimeSystemClock.asDateTimeLocal()));
};

JkTimeFormatVerboseDateTimeString.forDateTime = function(dd) {
	if(!(dd != null)) {
		return "NODATE";
	}
	var sb = JkLangStringBuilder.NEW();
	sb.appendString((JkTimeFormatVerboseDateTimeString.getShortDayName((dd.getWeekDay()))));
	sb.appendString(", ");
	sb.appendString((JkLangString.forInteger((dd.getDayOfMonth()))));
	sb.appendCharacter(' '.charCodeAt());
	sb.appendString((JkTimeFormatVerboseDateTimeString.getShortMonthName((dd.getMonth()))));
	sb.appendCharacter(' '.charCodeAt());
	sb.appendString((JkLangString.forInteger((dd.getYear()))));
	sb.appendCharacter(' '.charCodeAt());
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getHours()), 2, "0")));
	sb.appendCharacter(':'.charCodeAt());
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getMinutes()), 2, "0")));
	sb.appendCharacter(':'.charCodeAt());
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getSeconds()), 2, "0")));
	var offset = dd.getTimezoneOffset();
	if(offset == 0) {
		sb.appendString(" +0000");
	}
	else {
		var hr = ~(~(offset / 3600));
		var mm = ~(~((offset - hr * 3600) / 60));
		if(hr <= 0) {
			sb.appendString(" +");
			hr = -hr;
		}
		else {
			sb.appendString(" -");
		}
		sb.appendString((JkLangString.forIntegerWithPadding(hr, 2, "0")));
		sb.appendString((JkLangString.forIntegerWithPadding(mm, 2, "0")));
	}
	return sb.toString();
};

JkTimeFormatVerboseDateTimeString.getTimeStringForDateTime = function(dd, includeTimeZone) {
	if(!(dd != null)) {
		return "NOTIME";
	}
	var sb = JkLangStringBuilder.NEW();
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getHours()), 2, "0")));
	sb.appendCharacter(':'.charCodeAt());
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getMinutes()), 2, "0")));
	sb.appendCharacter(':'.charCodeAt());
	sb.appendString((JkLangString.forIntegerWithPadding((dd.getSeconds()), 2, "0")));
	if(includeTimeZone) {
		sb.appendString(" GMT");
	}
	return sb.toString();
};

JkTimeFormatVerboseDateTimeString.getDateStringForDateTime = function(dd) {
	if(!(dd != null)) {
		return "NODATE";
	}
	var sb = JkLangStringBuilder.NEW();
	sb.appendString((JkTimeFormatVerboseDateTimeString.getLongMonthName((dd.getMonth()))));
	sb.appendCharacter(' '.charCodeAt());
	sb.appendString((JkLangString.forInteger((dd.getDayOfMonth()))));
	sb.appendString(", ");
	sb.appendString((JkLangString.forInteger((dd.getYear()))));
	return sb.toString();
};

JkTimeFormatVerboseDateTimeString.getShortDayName = function(n) {
	switch(n) {
		case 1:
			return "Sun";
		case 2:
			return "Mon";
		case 3:
			return "Tue";
		case 4:
			return "Wed";
		case 5:
			return "Thu";
		case 6:
			return "Fri";
		case 7:
			return "Sat";
	}
	return null;
};

JkTimeFormatVerboseDateTimeString.getLongDayName = function(n) {
	switch(n) {
		case 1:
			return "Sunday";
		case 2:
			return "Monday";
		case 3:
			return "Tuesday";
		case 4:
			return "Wednesday";
		case 5:
			return "Thursday";
		case 6:
			return "Friday";
		case 7:
			return "Saturday";
	}
	return null;
};

JkTimeFormatVerboseDateTimeString.getShortMonthName = function(n) {
	switch(n) {
		case 1:
			return "Jan";
		case 2:
			return "Feb";
		case 3:
			return "Mar";
		case 4:
			return "Apr";
		case 5:
			return "May";
		case 6:
			return "Jun";
		case 7:
			return "Jul";
		case 8:
			return "Aug";
		case 9:
			return "Sep";
		case 10:
			return "Oct";
		case 11:
			return "Nov";
		case 12:
			return "Dec";
	}
	return null;
};

JkTimeFormatVerboseDateTimeString.getLongMonthName = function(n) {
	switch(n) {
		case 1:
			return "January";
		case 2:
			return "February";
		case 3:
			return "March";
		case 4:
			return "April";
		case 5:
			return "May";
		case 6:
			return "June";
		case 7:
			return "July";
		case 8:
			return "August";
		case 9:
			return "September";
		case 10:
			return "October";
		case 11:
			return "November";
		case 12:
			return "December";
	}
	return null;
};

JkTimeFormatVerboseDateTimeString.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkTimeFormatVerboseDateTimeString"] === true;
};
